patroni/postgresql/config.py

```diff
@@ -824,12 +824,19 @@ def get_server_parameters(self, config):
         parameters.update(cluster_name=self._postgresql.scope, listen_addresses=listen_addresses, port=str(port))
         if config.get('synchronous_mode', False):
             if self._synchronous_standby_names is None:
-                if config.get('synchronous_mode_strict', False):
+                if config.get('synchronous_mode_strict', False) and \
+                   hasattr(self._postgresql, 'role') and \
+                   self._postgresql.role == 'master':
                     parameters['synchronous_standby_names'] = '*'
                 else:
                     parameters.pop('synchronous_standby_names', None)
             else:
+                # TODO identify change in synchronous_node_count and compare it with current num_sync,
+                # adjust synchronous_standby_names accordingly. Leaving to existing value for now for
+                # Patroni to reset the value during next run cycle
                 parameters['synchronous_standby_names'] = self._synchronous_standby_names
+        else:
+            parameters.pop('synchronous_standby_names', None)
```

Comment:
During the tests, noticed that Patroni updating synchronous_standby_names on standby databases when synchronous_mode_strict was turned on. This change is to avoid changes on standby nodes.